lib/puppet/pops/loaders.rb

```diff
@@ -102,6 +106,16 @@ def self.loaders
     loaders
   end
 
+  def [](loader_name)
+    loader = @loaders_by_name[loader_name]
+    if loader.nil?
+      # Unable to find the module private loader. Try resolving the module
+      loader = private_loader_for_module(loader_name[0..-9]) if loader_name.end_with?(' private')
```

Comment:
This seems a bit brittle as it depends on an undocumented (and not enforced) naming convention.